test/src/e2e_vm_tests/test_programs/should_pass/stdlib/vec/src/main.sw

```diff
@@ -409,24 +817,233 @@ fn test_vector_new_enum() {
         },
         Option::None => revert(0), 
     }
-}
 
-fn test_vector_new_tuple() {
-    let mut vector = ~Vec::new();
+    vector.push(SimpleEnum::Y(b1));
+    vector.push(SimpleEnum::Y(b2));
```

Comment:
You've pushed 2 elements so you should assert the before condition again to ensure that the removal works correctly here too